community-operators/jaeger/jaeger.v1.11.0.clusterserviceversion.yaml

```diff
@@ -0,0 +1,300 @@
TkuNTUgMzM5LjUgOTkuOTUgMzM5LjQgMTAwLjIgMzM5LjE1CkwgMTAwLjMgMzM5LjE1ClEgMTAyLjM1IDMzOC45IDEwNC40IDMzOC43NSAxMDYuMjUgMzM4LjY1IDEwNy44IDMzOC4zCkwgMTA3LjggMzM4LjI1ClEgMTA3Ljc1IDMzOC4yIDEwNy44IDMzOC4wNSAxMDkuMjUgMzM4LjYgMTExLjI1IDMzOC4yNSAxMjAuNjUgMzM2LjYgMTMwLjMgMzM1LjIgMTMxLjg1IDMzNSAxMzMuNSAzMzQuNzUKTCAxMzMuNSAzMzQuNwpRIDEzNS4yNSAzMzMuODUgMTM3Ljc1IDMzMy44IDEzOC4zIDMzMy4zIDEzOS4zNSAzMzMuMzUgMTM5LjY1IDMzMy40IDEzOS45IDMzMy4zNSAxNDIuNzUgMzMyLjkgMTQ1LjA1IDMzMS44NSAxNDUuMDUgMzMxLjggMTQ1LjEgMzMxLjc1IDE0NS40NSAzMzEuNDUgMTQ1LjU1IDMzMC44NSAxNDUuNTUgMzMwLjggMTQ1LjUgMzMwLjc1IDE0NC40IDMyOS40NSAxNDIuNSAzMjguOSAxMzcuNyAzMjguOCAxMzMgMzI4LjQ1IDEyOC41NSAzMjguMSAxMjQuMSAzMjcuNzUgMTE4Ljg1IDMyNy4zNSAxMTMuNTUgMzI3LjcgMTA3IDMyOC42IDEwMi4zNSAzMzEuMzUgOTguODUgMzMzLjM1IDk1LjMgMzM1LjI1IDk1LjI1IDMzNS4yNSA5NS4xNSAzMzUuMjUgOTIuNyAzMzUuNyA5MC42NSAzMzYuNgpMIDkwLjY1IDMzNi42NQpRIDkxLjk1IDMzNi4zIDkzLjIgMzM2LjQ1IDkzLjE1IDMzNi40NSA5My4xIDMzNi41IDg5LjUgMzM3LjcgODUuOCAzMzguOSA4Ni42NSAzMzcuOTUgODguMSAzMzcuNjUgODkuNSAzMzcuMzUgOTAuNTUgMzM2LjcgOTAuNSAzMzYuNyA5MC40IDMzNi43IDg1Ljk1IDMzNy41NSA4Mi42NSAzMzkuNSA4Mi41NSAzMzkuNSA4Mi41IDMzOS41IDgwLjEgMzM5LjYgNzguMzUgMzQwLjM1IDc4LjMgMzQwLjM1IDc4LjI1IDM0MC4zNSA3NyAzNDAuMzUgNzYuMyAzNDAuODUgNzQuMTUgMzQxLjIgNzIuMDUgMzQxLjU1IFoiLz4KCjxwYXRoIGZpbGw9IiMyOTI5MjkiIGZpbGwtb3BhY2l0eT0iMC41NTI5NDExNzY0NzA1ODgzIiBzdHJva2U9Im5vbmUiIGQ9IgpNIDYwLjIgMzQzLjg1ClEgNTIuMyAzNDUuNTUgNDMuNTUgMzQ2LjMKTCA0My41NSAzNDYuNApRIDM5LjcgMzQ3LjMgMzguMSAzNTAuMzUgMzguMDUgMzUxLjIgMzguNCAzNTEuNzUgMzguNCAzNTEuOCAzOC40NSAzNTEuOCAzOC40NSAzNTEuNzUgMzguNDUgMzUxLjY1IDM4LjM1IDM1MS4xIDM4LjM1IDM1MC40NSAzOS43NSAzNDcuNDUgNDMuNDUgMzQ2LjcgNDMuNSAzNDYuNyA0My41NSAzNDYuNyA0OC4xIDM0Ni4xIDUyLjcgMzQ1LjY1IDU3LjggMzQ1LjIgNjEuODUgMzQzLjkgNjYuNjUgMzQyLjM1IDcyLjA1IDM0MS41NSA3NC4xNSAzNDEuMiA3Ni4zIDM0MC44NSA3NyAzNDAuMzUgNzguMjUgMzQwLjM1IDc4LjMgMzQwLjM1IDc4LjM1IDM0MC4zNSA4MC4xIDMzOS42IDgyLjUgMzM5LjUgODIuNTUgMzM5LjUgODIuNjUgMzM5LjUgODUuOTUgMzM3LjU1IDkwLjQgMzM2LjcgOTAuNSAzMzYuNyA5MC41NSAzMzYuNyA4OS41IDMzNy4zNSA4OC4xIDMzNy42NSA4Ni42NSAzMzcuOTUgODUuOCAzMzguOSA4OS41IDMzNy43IDkzLjEgMzM2LjUgOTMuMTUgMzM2LjQ1IDkzLjIgMzM2LjQ1IDkxLjk1IDMzNi4zIDkwLjY1IDMzNi42NQpMIDkwLjY1IDMzNi42ClEgOTIuNyAzMzUuNyA5NS4xNSAzMzUuMjUgOTUuMjUgMzM1LjI1IDk1LjMgMzM1LjI1IDk4Ljg1IDMzMy4zNSAxMDIuMzUgMzMxLjM1IDEwNyAzMjguNiAxMTMuNTUgMzI3LjcgMTEzLjU1IDMyNy42NSAxMTMuNTUgMzI3LjYgMTE4Ljg1IDMyNy4xNSAxMjMuOSAzMjcuNTUgMTI4Ljc1IDMyNy45NSAxMzMuNzUgMzI4LjQ1IDEzNC42NSAzMjguNSAxMzQuODUgMzI4LjMgMTM1LjIgMzI3LjcgMTM2LjcgMzI3LjkgMTM5LjMgMzI4LjIgMTQxLjggMzI4LjM1IDE0NC40IDMyOC41NSAxNDUuOSAzMjkuNzUgMTQ2LjYgMzMxIDE0NS45NSAzMzIuMiAxNDUuOSAzMzIuMjUgMTQ1LjkgMzMyLjM1IDE0NS41NSAzMzIuNjUgMTQ1LjEgMzMzIDE0NS4wNSAzMzMgMTQ1LjA1IDMzMy4wNSAxNDUuMSAzMzMuMDUgMTQ1LjE1IDMzMy4wNSAxNDYuOCAzMzIuMjUgMTQ2LjQgMzMwIDE0Ni4zNSAzMzAgMTQ2LjMgMzMwIDE0Ni4wNSAzMjkuNSAxNDUuNjUgMzI5LjE1IDE0Mi4yIDMyOC4wNSAxMzggMzI3Ljg1IDEzNS4zNSAzMjcuNyAxMzMgMzI3LjU1IDEyNy45NSAzMjcuMyAxMjIuOCAzMjYuOTUgMTE4LjkgMzI2Ljc1IDExNS4xIDMyNy4wNSAxMDcuNCAzMjcuNjUgMTAyLjI1IDMzMC42IDEwMC41IDMzMS42NSA5OC44NSAzMzIuNiA5NS4zNSAzMzQuNjUgOTEuM
```

Comment:
The `WATCH_NAMESPACE` environment variable should be set to the following:
```
valueFrom:
  fieldRef:
    fieldPath: metadata.annotations['olm.targetNamespaces']
```
This is required for OLM to be able to inform operators of which namespaces they should watch. I believe Jaeger supports the downward API for watching all namespaces (using the empty string to represent AllNamespaces), based on previous versions.

Reply:
Updated.

Reply:
The Jaeger Operator should be able to watch other namespaces than the one it's deployed. I might have done something wrong, but in my local tests it seems that this annotation has the value of `olm`, meaning that the operator will not process any `Jaeger` instances deployed into other namespaces, like `observability`.

Reply:
@jpkrohling Depending on what OperatorGroup is present in the namespace the operator is installed to, this annotation will be set appropriately. If the operator is installed into the `olm` namespace, then the OperatorGroup present there by default will be used, which watches the olm namespace. If you install the operator to the `operators` namespace (on a Kubernetes cluster with OLM installed), the `global-operators` OperatorGroup will have this annotation set to the empty string to inform the operator to watch all namespaces clusterwide. Since this operator supports both SingleNamespace and AllNamespaces installations, it should be able to either watch one namespace or all namespaces, depending on how this annotation is set.
More info on OperatorGroups: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/operatorgroups.md

Reply:
@objectiser I'm not sure if the change went through

Reply:
Thanks for the clarification!